docs/guides/reproducible-builds.rst

```diff
@@ -0,0 +1,179 @@
+- :doc:`/intro/getting-started-with-mkdocs`
+- :doc:`/config-file/v2`
+
+.. note::
+
+   You can see the exact commands that are run on Read the Docs by going to the :guilabel:`Builds` tab of your project.
+
+Using a configuration file
+--------------------------
+
+If you use the web interface to configure your project,
+the options are applied to *all* versions and builds of your docs,
+and can be lost after changing them over time.
+Using a :doc:`configuration file </config-file/v2>` **provides you per version settings**,
+and **those settings live in your repository**.
+
+A configuration file with explicit dependencies looks like this:
+
+.. code-block:: yaml
+   
+   # File: readthedocs.yaml
+
+   version: 2
+
+   # Build from the docs/ directory with Sphinx
+   sphinx:
+     configuration: docs/conf.py
+
+   # Explicitly set the version of Python and its requirements
+   python:
+     version: 3.7
+     install:
+       - requirements: docs/requirements.txt
+
+.. code-block::
+
+   # File: docs/requirements.txt
+
+   # Explicitly dependencies and versions
+   sphinx==3.4.3
+   sphinx_rtd_theme==0.5.1
+   readthedocs-sphinx-search==0.1.0rc3
+
+Don't rely on implicit dependencies
+-----------------------------------
+
+By default Read the Docs will install the tool you chose to build your docs,
+and other dependencies, this is done so new users can build their docs without much configuration.
+
+We highly recommend not to assume these dependencies will always be present or that their versions won't change.
+Always declare your dependencies explicitly using a :ref:`configuration file <guides/reproducible-builds:using a configuration file>`,
+for example:
+
+✅ Good:
+   Your project is declaring the Python version explicitly,
+   and its dependencies using a requirements file.
+
+   .. code-block:: yaml
+      
+      # File: .readthedocs.yaml
+
+      version: 2
+
+      sphinx:
+        configuration: docs/conf.py
+
+      python:
+        version: 3.7
+        install:
+          - requirements: docs/requirements.txt
+
+❌ Bad:
+   Your project is relying on the default Python version and default installed dependencies.
+
+   .. code-block:: yaml
+      
+      # File: .readthedocs.yaml
+
+      version: 2
+
+      sphinx:
+         configuration: docs/conf.py
+
+Pinning dependencies
+--------------------
+
+As you shouldn't rely on implicit dependencies,
+you shouldn't rely on arbitrary versions of your dependencies.
+Some examples:
+
+✅ Good:
+   The specified versions will be used for all your builds,
+   in all platforms, and won't be updated unexpectedly.
+
+   .. code-block::
+
+      # File: docs/requirements.txt
+
+      sphinx==3.4.3
+      sphinx_rtd_theme==0.5.1
+      readthedocs-sphinx-search==0.1.0rc3
+
+   .. code-block:: yaml
+      
+      # File: docs/environment.yaml
+
+      name: docs
+      channels:
+        - conda-forge
+        - defaults
+      dependencies:
+        - sphinx==3.4.3
```

Comment:
If I understand correctly, core dependencies can't be pinned in conda environment files at the moment, right? https://github.com/readthedocs/readthedocs.org/issues/3829

Reply:
I just tried it again to be sure, Sphinx gets overridden https://readthedocs.org/projects/rtd-conda-test/builds/13122163/

Reply:
They _can_ be pinned. However, `CONDA_APPEND_CORE_REQUIREMENTS` feature flag has to be enabled in the project by the core team: https://docs.readthedocs.io/en/stable/guides/feature-flags.html#feature-flags

Reply:
BTW, for future build tests, we have this repository https://github.com/readthedocs/test-builds where we have one test case per branch (properly named) with the use case that it's testing described in the homepage/index :wink: 

(you can create a PR instead of a branch since you don't have permissions)

Reply:
> They can be pinned. However, CONDA_APPEND_CORE_REQUIREMENTS feature flag has to be enabled in the project by the core team

I'm tempted to add a note about that here, but I think maybe we should push to make that the default.